packages/subproviders/src/subproviders/private_key_wallet.ts

```diff
@@ -42,26 +45,33 @@ export class PrivateKeyWalletSubprovider extends BaseWalletSubprovider {
      */
     public async signTransactionAsync(txParams: PartialTxParams): Promise<string> {
         PrivateKeyWalletSubprovider._validateTxParams(txParams);
+        if (!_.isUndefined(txParams.from) && txParams.from !== this._address) {
+            throw new Error(`${WalletSubproviderErrors.AddressNotFound}: ${txParams.from}`);
```

Comment:
I think we need to throw a better error message here. Perhaps: 

```
Requested to sign transaction with account: ${txParams.from} using PrivateKeyWalletSubprovider instantiated with account: ${this._address}
```

Reply:
Sure, since this code is re-used by the Mnemonic wallet as well I think we should use the message you provided without mention of PrivateKeyWalletSubprovider.